app/src/utils/data/contributions.ts

```diff
@@ -368,3 +375,103 @@ export function grantDonationListener(
     off: () => grantRoundManager.value.off(filter, listener),
   };
 }
+
+/**
+ * @notice given a userAddress and the contributions, return only the contributions for that address
+ * @param userAddress
+ * @param contributions
+ */
+export function filterContributionsByUserAddress(userAddress: string, contributions: Contribution[]) {
+  if (contributions && userAddress) {
+    return contributions.filter((contribution: Contribution) => {
+      return contribution.address === userAddress;
+    });
+  }
+  return undefined;
+}
+
+/**
+ * @notice filters through the grantrounds and returns the necessary attributes for  contribution detail screens
+ * @param grantRounds
+ * @param grantId
+ * @param grantRoundMeta
+ */
+function filterGrantRoundsForContributions(
+  grantRounds: GrantRound[],
+  grantId: number,
+  grantRoundMeta?: Record<string, GrantRoundMetadata>
+) {
+  let roundName = '...';
+
+  if (grantRoundMeta) {
```

Comment:
Since `Boolean({})` resolves to true, do we need a check that `grantRoundMeta` is not an empty object? Something like`Object.keys(grantRoundMeta).length > 0`

Reply:
so you are I are correct! Just need to combine  - `if (grantRoundMeta && Object.keys(grantRoundMeta).length > 0) ` because if grantRoundMeta does not exist then the Object.keys will freak out